reconcilers/stash.go

```diff
@@ -55,3 +56,72 @@ func ClearValue(ctx context.Context, key StashKey) interface{} {
 	delete(stash, key)
 	return value
 }
+
+// Stasher stores and retrieves values from the stash context. The context must be configured
+// with a stash via WithStash(). The stash is pre-configured for the context within a reconciler.
+type Stasher[T any] interface {
+	// Key is the stash key used to store and retrieve the value
+	Key() StashKey
+
+	// Store saves the value in the stash under the key
+	Store(ctx context.Context, value T)
```

Comment:
**[thought]:** _Store_ is a good name. However, intuitively I was expecting it to be _Stash_. That's in the spirit of _Stringer, Writer, Reader_ etc.

Reply:
That makes sense to me @scothis! ty